anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/BasicAnomalyDetectorFactory.java

```diff
@@ -42,41 +43,32 @@
 @Slf4j
 public final class BasicAnomalyDetectorFactory<T extends BasicAnomalyDetector> implements AnomalyDetectorFactory<T> {
     private Class<T> detectorClass;
-    private String region;
-    private String bucket;
-    private String folder;
-    
-    private AmazonS3 s3;
+
     private final ObjectMapper objectMapper = new ObjectMapper();
-    
+
     @Override
     public void init(Config config) {
         notNull(config, "config can't be null");
-    
+
         this.detectorClass = ReflectionUtil.classForName(config.getString("detectorClass"));
-        this.region = config.getString("region");
-        this.bucket = config.getString("bucket");
-        this.folder = config.getString("folder");
-        
-        this.s3 = AmazonS3ClientBuilder.standard()
-                .withRegion(region)
-                .build();
-        
-        log.info("Initialized factory: detectorClass={}, region={}, bucket={}, folder={}",
-                detectorClass.getName(), region, bucket, folder);
+        log.info("Initialized factory: detectorClass={}", detectorClass.getName());
     }
-    
+
     @Override
-    public T create(UUID uuid) {
+    public T create(UUID uuid, ModelServiceConnector modelServiceConnector) {
         notNull(uuid, "uuid can't be null");
-        
-        final String path = folder + "/" + uuid.toString() + ".json";
-        log.info("Loading model: path={}", path);
-        final S3Object s3Obj = s3.getObject(bucket, path);
-        final InputStream is = s3Obj.getObjectContent();
-        final AnomalyDetectorModel model = ObjectMapperUtil.readValue(objectMapper, is, AnomalyDetectorModel.class);
-        log.info("Loaded model: {}", model);
-        
+
+        final Resources<ModelResource> modelResources = modelServiceConnector.findModels(uuid);
+        final Collection<ModelResource> modelResourceCollection = modelResources.getContent();
+
+        List<ModelResource> modelResourceList = new ArrayList<>(modelResourceCollection);
+
+        if (modelResourceList.size() < 1) {
+            log.error("No model found for uuid = {}", uuid);
+        }
+        final AnomalyDetectorModel model = modelResourceList.get(0);
```

Comment:
Compare this to the previous block where you check for modelResourceList.size() < 1. If that condition obtains then you will get an IndexOutOfBoundsException here. Decide how you want to handle the case where modelResourceList.size() < 1 and then make sure the code supports it.